commons-math-core/src/main/java/org/apache/commons/math4/core/jdkmath/AccurateMath.java

```diff
@@ -1176,8 +1190,10 @@ private static double log(final double x, final double[] hiPrec) {
             }
         }
 
+        boolean isExpNegOneOrZero = (exp == -1 || exp == 0);
```

Comment:
Use of variables as comments. There are already comments explaining the choice of x in [0.99, 1.01].

Note the exponent is checked for -1 or 0 first to capture x in [0.5, 2). I presume this is faster than a direct check on the magnitude of x. If this is true, then the magnitude of x is checked within the range. So moving the checks to variables removes the ability to skip the magnitude check as storing in variables means both computations are always done.